### PR TITLE
Internal: Improved element regression screenshots [ED-11627]

### PR DESCRIPTION
**Pull request by dmaziar**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:
Instead of screenshot for every widget in editor, the test will do one editor screenshot, the same as after publish.
It is now possible, because I increased `iframe.style.height` and viewport in config `viewport: { width: 1920, height: 3080 }`. This helps to remove a lot of screenshots and increase execution time.
Screenshot: 266  before, 165 after

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: tests/elements-regression/helper.ts [hunk 1]
```diff
@@ -3,10 +3,10 @@ import EditorSelectors from '../playwright/selectors/editor-selectors';
 import { expect, type Frame, type Page, type TestInfo } from '@playwright/test';
 type ScreenShot = {
 	device: string,
-	isPublished?: boolean,
-	widgetType: string
+	isPublished: boolean,
+	widgetType: string,
+	hoverSelector: {[ key: string ]: string}
```
Comment (davseve) [inline tests/elements-regression/helper.ts:8]:
```suggestion
	hoverSelector: { [ key: string ]: string }
```

Comment (davseve) [inline tests/elements-regression/helper.ts:8]:
```suggestion
	hoverSelector: { [ key: string ]: string }
```

File: tests/elements-regression/helper.ts [hunk 2]
```diff
@@ -16,37 +16,44 @@ export default class ElementRegressionHelper {
...
 	}
 
-	async doScreenshotPublished( args: { widgetType: string, hoverSelector: {[ key: string ]: string} } ) {
-		const widgetCount = await this.page.locator( EditorSelectors.widget ).count();
+	async doHoverScreenshot( args:Omit<ScreenShot, 'device'> ) {
 		if ( args.widgetType.includes( 'hover' ) ) {
+			const widget = args.isPublished
+				? this.page.locator( EditorSelectors.widget )
+				: this.editorPage.getPreviewFrame().locator( EditorSelectors.widget );
+			const label = args.isPublished ? 'published' : 'editor';
```
Comment (davseve) [inline tests/elements-regression/helper.ts:44]: I see that this function appears more than once, 
Maybe we can locate it in a central location.

File: tests/elements-regression/helper.ts [hunk 3]
```diff
@@ -16,37 +16,44 @@ export default class ElementRegressionHelper {
...
+			const widgetCount = await widget.count();
+
 			for ( let i = 0; i < widgetCount; i++ ) {
-				await this.page.locator( `${ EditorSelectors.widget } ${ args.hoverSelector[ args.widgetType ] }` ).nth( i ).hover();
-				await expect.soft( this.page.locator( EditorSelectors.widget ).nth( i ) ).
-					toHaveScreenshot( `${ args.widgetType }_${ i }_published.png`, { maxDiffPixels: 200, timeout: 10000, animations: 'allow' } );
+				await expect( widget.nth( i ) ).not.toHaveClass( /elementor-widget-empty/ );
```
Comment (davseve) [inline tests/elements-regression/helper.ts:48]: Do we want to have a soft assertion here?
```suggestion
				await expect.soft( widget.nth( i ) ).not.toHaveClass( /elementor-widget-empty/ );
```

File: tests/elements-regression/tests/elements-regression.test.ts [hunk 4]
```diff
@@ -68,15 +68,19 @@ test.describe( 'Elementor regression tests with templates for CORE', () => {
 
 			await editorPage.loadTemplate( filePath, true );
 			await editorPage.waitForIframeToLoaded( widgetType );
-			await helper.doScreenshotComparison( { widgetType, hoverSelector } );
+
+			await page.setViewportSize( { width: 1920, height: 3080 } );
```
Comment (davseve) [inline tests/elements-regression/tests/elements-regression.test.ts:72]: I see that this viewport size is pretty unique,
But maybe we can add the screen sizes to the `viewportSize` const in `tests/playwright/assets/elements-utils.ts`, that way we will have one source of truth to all screen sizes.

File: tests/elements-regression/helper.ts (hunk 3 above)
Comment (dmaziar) [reply tests/elements-regression/helper.ts:48]: I don't want to continue if widget failed to load, so no soft here

File: tests/elements-regression/tests/elements-regression.test.ts (hunk 4 above)
Comment (dmaziar) [reply tests/elements-regression/tests/elements-regression.test.ts:72]: this case is pretty uniq. I need this big viewport in order to do a big screenshot with more then one widget. In all other place it's not needed.

File: tests/elements-regression/helper.ts (hunk 1 above)
Comment (dmaziar) [reply tests/elements-regression/helper.ts:8]: fixed

File: tests/elements-regression/helper.ts (hunk 2 above)
Comment (dmaziar) [reply tests/elements-regression/helper.ts:44]: fixed

**Review by davseve — DISMISSED** [review]

**Review by hein-obox — APPROVED** [review]

**Pull request merged**